sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -759,6 +759,12 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
         def entry(key: String, value: Token): Seq[(String, String)] = {
           Option(value).map(t => key -> t.getText).toSeq
         }
+
+        if (Option(c.linesSeparatedBy).map(string).getOrElse("\n") != "\n") {
```

Comment:
We need to follow this restriction of Hive? IMO its okay to support any char as a line separator normally.

Reply:
> IMO its okay to support any char as a line separator normally.

In this way ,  line Separator must be a char. since it's hard to write a high-performance Reader (like BufferedReader.getLine) that can get one row data with string line separator . 

if just support char as line delim, we can just rewrite a reader like BufferedReader.

> We need to follow this restriction of Hive?
 Script transform's user always come from hive. It ok just to follow hive and this behavior is straight forward and simple.

Reply:
cc @maropu  https://github.com/apache/spark/blob/1a4c8f718f748d8c2a39f0a3b209ce606bffe958/sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala#L671-L675